core/src/tile/tileTask.cpp

```diff
@@ -9,4 +9,10 @@ TileTask::TileTask(TileID& _tileId, std::shared_ptr<DataSource> _source) :
     m_sourceGeneration(_source->generation()),
     m_priority(0) {}
 
+void TileTask::cancel() {
+    for (auto& raster : m_rasterTasks) {
+        raster->cancel();
+    }
+    m_canceled = true;
```

Comment:
```m_rasterTasks``` can be modified concurrently (though with an almost minimal chance) when TileTask::cancel() is called on the TileWorker while m_rasterTasks are added or canceld in TileManager update